product_move_analytical_account/hooks.py

```diff
@@ -0,0 +1,14 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+import logging
+from odoo import SUPERUSER_ID, api, _
+
+
+def post_init_hook(cr, registry):
+    logging.getLogger('odoo.addons.interface_gaia').info(_('Adding Account Analytic from old purchase order line to stock.move.line'))
+
+    env = api.Environment(cr, SUPERUSER_ID, {})
+
+    for record in env["stock.move.line"].search([('account_analytic_id', '=', False)]):
+        record.account_analytic_id = record.move_id.purchase_line_id.account_analytic_id
```

Comment:
Why not this?

```suggestion
    move_lines = env["stock.move.line"].search([('move_id.purchase_line_id.account_analytic_id', '!=', False)]):
    move_lines._compute_account_analytic()
```

Anyway, when the compute stored field is created (during addon installation) it should be automatically filled, so hook is not needed (at least in other Odoo versions it works). Could you check?

Reply:
I tested It again and finally, is not necessary to create a hook to fill old fields, with computed store=true is enought.